ext/compressors/lz4/lz4_compress.c

```diff
@@ -277,8 +277,9 @@ lz4_compress_raw(WT_COMPRESSOR *compressor, WT_SESSION *session,
 	(void)final;
 
 	sourceSize = (int)offsets[slots];		/* Type conversion */
-	targetDestSize =
-	    (int)((dst_len < page_max ? dst_len : page_max) - extra);
+	targetDestSize = (int)
+	    (((dst_len - sizeof(LZ4_PREFIX)) < page_max ?
+	    (dst_len - sizeof(LZ4_PREFIX)) : page_max) - extra);
```

Comment:
I'm struggling to parse this and when I do I'm not sure it's right.

`zlib_compress_raw` ignores `dst_len` completely.  Why is that okay, but lz4 checks it?

If we have to do something here, how about `extra += sizeof(LZ4_PREFIX)` before this?